django/db/models/fields/related.py

```diff
@@ -2051,6 +2052,18 @@ def db_type(self, connection):
     def db_parameters(self, connection):
         return {"type": self.db_type(connection), "check": []}
 
+    def get_db_converters(self, connection):
+        backend_converters = connection.ops.get_db_converters(Col(self.attname, self.related_field))
+        backend_converters = [self._convert_backend_converter(func) for func in backend_converters]
+        field_converters = self.related_field.get_db_converters(connection)
+        self_converters = super(ForeignKey, self).get_db_converters(connection)
```

Comment:
Adding here these lines fixes the failures:
```python
        if connection.features.interprets_empty_strings_as_nulls:
            self_converters.append(lambda val,*args: 
                                    None if isinstance(val, six.string_types) and not val 
                                    else val)
```
but this is getting uglier and uglier